### PR TITLE
Update flow-bin to version 0.64.0

### PR DESCRIPTION
**Pull request by Arcanemagus**
Updates `flow-bin` to version 0.64.0.

Fix the new Flow issues exposed by the updated version. Mostly this just entailed adding explicit types to the `React.Component` inheritance.

Closes #419.

### CODE REVIEW

**Comment by Arcanemagus** [comment]
**NOTE**: Based off of the branch for #472, this will need to be changed to base on `master` before merging.

File: lib/index.js [hunk 1]
```diff
@@ -3,9 +3,15 @@
 import LinterUI from './main'
 import type Intentions from './intentions'
 
-const idleCallbacks = new Set()
+const idleCallbacks: Set<any> = new Set()
```
Comment (steelbrain) [inline lib/index.js]: Could we make this `Set<number>` please?

File: lib/panel/component.js [hunk 2]
```diff
@@ -6,7 +6,15 @@ import { $range, severityNames, sortMessages, visitMessage, openExternally, getP
 import type Delegate from './delegate'
 import type { LinterMessage } from '../types'
 
-class PanelComponent extends React.Component {
+type Props = {
```
Comment (steelbrain) [inline lib/panel/component.js]: It's generally a good idea to do enforced type objects, like
```
type Props = {|
  delegate: Delegate,
|}
```

Normally flows allows *extra* properties in a typed object, this makes sure it doesn't

File: lib/panel/component.js [hunk 3]
```diff
@@ -6,7 +6,15 @@ import { $range, severityNames, sortMessages, visitMessage, openExternally, getP
...
+  delegate: Delegate,
+}
+
+type State = {
```
Comment (steelbrain) [inline lib/panel/component.js]: Ditto about enforced object type

File: lib/panel/component.js [hunk 4]
```diff
@@ -57,9 +66,9 @@ class PanelComponent extends React.Component {
       visitMessage(row)
     }
   }
-  props: {
-    delegate: Delegate,
-  };
+
+  props: Props;
```
Comment (steelbrain) [inline lib/panel/component.js]: You can remove this declaration because it'll be inherited now

File: lib/tooltip/message-legacy.js [hunk 5]
```diff
@@ -7,10 +7,17 @@ import type { MessageLegacy } from '../types'
 const NEWLINE = /\r\n|\n/
 let MESSAGE_NUMBER = 0
 
-class MessageElement extends React.Component {
-  state: {
-    multiLineShow: boolean,
-  } = {
+type Props = {
```
Comment (steelbrain) [inline lib/tooltip/message-legacy.js]: Enforced object type here and below please

File: lib/tooltip/message-legacy.js [hunk 6]
```diff
@@ -24,10 +31,7 @@ class MessageElement extends React.Component {
       this.setState({ multiLineShow: false })
     })
   }
-  props: {
-    message: MessageLegacy,
-    delegate: TooltipDelegate,
-  };
+  props: Props;
```
Comment (steelbrain) [inline lib/tooltip/message-legacy.js]: Ditto about removing the typing for props

File: lib/tooltip/message.js [hunk 7]
```diff
@@ -19,11 +19,18 @@ function findHref(el: ?Element): ?string {
   return null
 }
 
-class MessageElement extends React.Component {
-  state: {
-    description: string,
-    descriptionShow: boolean,
-  } = {
+type Props = {
```
Comment (steelbrain) [inline lib/tooltip/message.js]: and here

File: lib/tooltip/message.js [hunk 8]
```diff
@@ -117,10 +124,7 @@ class MessageElement extends React.Component {
     }
   }
 
-  props: {
-    message: Message,
-    delegate: TooltipDelegate,
-  };
+  props: Props;
```
Comment (steelbrain) [inline lib/tooltip/message.js]: and here 😛

File: lib/index.js (hunk 1 above)
Comment (Arcanemagus) [reply lib/index.js]: [The spec](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback) only specifies "identifier", it doesn't actually say what that identifier _is_, so `<any>` is appropriate here.

**Comment by Arcanemagus** [comment]
So unfortunately the types can't be restricted like that because the old `babylon` parser used inside Atom doesn't know about that syntax and fails to transpile the code. Removing that syntax allows the transpilation to finish... but then leads to [this output](https://circleci.com/gh/steelbrain/linter-ui-default/735). So it looks like Flow isn't smart enough to associate the types like that automatically (at least not without the restricted types).

I've reverted this branch to before the changes to address the review comments, since it seems none of them are able to be applied.

**Comment by Arcanemagus** [comment]
It looks like greenkeeper deleting the original base branch for this broke the PR. Recreated in #474.

**Pull request closed without merging**